recce/adapter/dbt_adapter/__init__.py

```diff
@@ -621,11 +621,15 @@ def execute(
 
     def build_parent_map(self, nodes: Dict, base: Optional[bool] = False) -> Dict[str, List[str]]:
         manifest = self.curr_manifest if base is False else self.base_manifest
-        manifest_dict = manifest.to_dict()
+        
+        try:
+            parent_map_source = manifest.parent_map
+        except AttributeError:
+            parent_map_source = manifest.to_dict()["parent_map"]
 
-        node_ids = nodes.keys()
+        node_ids = set(nodes)
```

Comment:
The keys view is set-like, so I think we don't need to change it to `set` to improve the lookup time.

Reply:
fair. making the edit